static/app/views/performance/transactionSummary/transactionOverview/durationPercentileChart/content.tsx

```diff
@@ -41,86 +33,72 @@ type State = DeprecatedAsyncComponent['state'] & {
  * This graph visualizes how many transactions were recorded
  * at each duration bucket, showing the modality of the transaction.
  */
-class Content extends DeprecatedAsyncComponent<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {
-      organization,
-      query,
-      start,
-      end,
-      statsPeriod,
-      environment,
-      project,
-      fields,
-      location,
-      queryExtras,
-    } = this.props;
-
-    const eventView = EventView.fromSavedQuery({
-      id: '',
-      name: '',
-      version: 2,
-      fields,
-      orderby: '',
-      projects: project,
-      range: statsPeriod,
-      query,
-      environment,
-      start,
-      end,
-    });
-    let apiPayload = eventView.getEventsAPIPayload(location);
-    apiPayload = {
-      ...apiPayload,
-      ...queryExtras,
-      referrer: 'api.performance.durationpercentilechart',
-    };
-    const endpoint = `/organizations/${organization.slug}/events/`;
-
-    return [['chartData', endpoint, {query: apiPayload}]];
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    if (this.shouldRefetchData(prevProps)) {
-      this.fetchData();
-    }
-  }
+function Content({
+  currentFilter,
+  fields,
+  location,
+  organization,
+  queryExtras,
+  start,
+  end,
+  query,
+  statsPeriod,
+  environment,
+  project,
+}: Props) {
+  const eventView = EventView.fromSavedQuery({
+    id: '',
+    name: '',
+    version: 2,
+    fields,
+    orderby: '',
+    projects: project,
+    range: statsPeriod,
+    query,
+    environment,
+    start,
+    end,
+  });
+  let apiPayload = eventView.getEventsAPIPayload(location);
+  apiPayload = {
+    ...apiPayload,
+    ...queryExtras,
+    referrer: 'api.performance.durationpercentilechart',
+  };
 
-  shouldRefetchData(prevProps: Props) {
```

Comment:
the only thing i was iffy about - the error state is specifically a warning icon to match the other charts so i couldn't use `LoadingError` here, so didn't know what to do with `refetch` to replace the `shouldRefetchData`